src/components/Select/Select.tsx

```diff
@@ -141,6 +141,13 @@ const Select: React.FunctionComponent<React.PropsWithChildren<SelectProps>> = ({
     }
   }, [])
 
+  useEffect(() => {
+    if(defaultOptionIndex) {
+      setSelectedOptionIndex(defaultOptionIndex - 1)
```

Comment:
start index by 0

Reply:
If the index starts by 0, we can't detect first changes when the user sort the table using table header.
That's why I starts the index by 1.

Reply:
```
<Text color={!optionSelected && placeHolderText ? 'text' : undefined}>
  {!optionSelected && placeHolderText ? placeHolderText : options[selectedOptionIndex].label}
</Text>
```
As you can see here, **select** component only shows the current sort field when optionSelected is false.
In case the user sort the table using table header, that value doesn't change.
I solved this problem by starting the index from 1.